android/app-example/src/test/java/com/badoo/ribs/example/rib/foo_bar/FooBarInteractorTest.kt

```diff
@@ -19,16 +33,52 @@ class FooBarInteractorTest {
             router = router,
             permissionRequester = permissionRequester
         )
+
+        nodeHelper = RibTestHelper.create(interactor, foobarViewEventEmitter, router)
     }
 
     @After
     fun tearDown() {
     }
 
+    @Test
+    fun `check permission when check permission button clicked`() {
+        val inputSubscriber = interactor.dummyViewInput.test()
+        val result = PermissionRequester.CheckPermissionsResult(emptyList(), emptyList(), emptyList())
+        whenever(permissionRequester.checkPermissions(any(), any())).doReturn(result)
+
+        nodeHelper.startAndExecute {
+            foobarViewEventEmitter.accept(CheckPermissionsButtonClicked)
+
+            inputSubscriber.assertLast(FooBarView.ViewModel(result.toString()))
+        }
+    }
+
+    @Test
+    fun `does not check permission when check permission button clicked event came after onStop`() {
+        val inputSubscriber = interactor.dummyViewInput.test()
+        val result = PermissionRequester.CheckPermissionsResult(emptyList(), emptyList(), emptyList())
+        whenever(permissionRequester.checkPermissions(any(), any())).doReturn(result)
+
+        nodeHelper.startAndExecute {}
```

Comment:
How about creating two sets of methods?
`start` and `startAndExecute`
Right now it looks a bit ambiguous how this things work